cpp/tests/experimental/rw_low_level_test.cu

```diff
@@ -0,0 +1,877 @@
+    auto delta = row_offsets[crt_vertex] + std::distance(begin, found_next);
+
+    // std::cout << "delta in ci: " << delta << '\n';
+    auto found_edge = values.begin() + delta;
+    if (*found_edge != *it_w) {
+      std::cout << "weight not found: " << *found_edge << " between " << crt_vertex << " and "
+                << next_vertex << '\n';
+      return false;
+    }
+  }
+  return true;
+}
+
+template <typename vertex_t, typename edge_t, typename weight_t, typename index_t = edge_t>
+bool host_check_rw_paths(raft::handle_t const& handle,
+                         graph_view_t<vertex_t, edge_t, weight_t, false, false> const& graph_view,
+                         vector_test_t<vertex_t> const& d_coalesced_v,
+                         vector_test_t<weight_t> const& d_coalesced_w,
+                         vector_test_t<index_t> const& d_sizes)
+{
+  edge_t num_edges      = graph_view.get_number_of_edges();
+  vertex_t num_vertices = graph_view.get_number_of_vertices();
+
+  edge_t const* offsets   = graph_view.offsets();
+  vertex_t const* indices = graph_view.indices();
+  weight_t const* values  = graph_view.weights();
+
+  std::vector<edge_t> v_ro(num_vertices + 1);
+  std::vector<vertex_t> v_ci(num_edges);
+  std::vector<weight_t> v_vals(num_edges);
+
+  raft::update_host(v_ro.data(), offsets, v_ro.size(), handle.get_stream());
+  raft::update_host(v_ci.data(), indices, v_ci.size(), handle.get_stream());
+  raft::update_host(v_vals.data(), values, v_vals.size(), handle.get_stream());
+
+  std::vector<vertex_t> v_coalesced(d_coalesced_v.size());
+  std::vector<weight_t> w_coalesced(d_coalesced_w.size());
+  std::vector<index_t> v_sizes(d_sizes.size());
+
+  raft::update_host(v_coalesced.data(),
+                    detail::raw_const_ptr(d_coalesced_v),
+                    d_coalesced_v.size(),
+                    handle.get_stream());
+  raft::update_host(w_coalesced.data(),
+                    detail::raw_const_ptr(d_coalesced_w),
+                    d_coalesced_w.size(),
+                    handle.get_stream());
+  raft::update_host(
+    v_sizes.data(), detail::raw_const_ptr(d_sizes), d_sizes.size(), handle.get_stream());
+
+  auto it_v_begin = v_coalesced.begin();
+  auto it_w_begin = w_coalesced.begin();
+  for (auto&& crt_sz : v_sizes) {
+    auto it_v_end = it_v_begin + crt_sz;
+
+    bool test_path = host_check_path(v_ro, v_ci, v_vals, it_v_begin, it_v_end, it_w_begin);
+
+    it_v_begin = it_v_end;
+    it_w_begin += crt_sz - 1;
+
+    if (!test_path) {  // something went wrong; print to debug (since it's random)
+      raft::print_host_vector("sizes", v_sizes.data(), v_sizes.size(), std::cout);
+
+      raft::print_host_vector("coalesced v", v_coalesced.data(), v_coalesced.size(), std::cout);
+
+      raft::print_host_vector("coalesced w", w_coalesced.data(), w_coalesced.size(), std::cout);
+
+      return false;
+    }
+  }
+  return true;
+}
+}  // namespace
+
+struct RandomWalksPrimsTest : public ::testing::Test {
+};
```

Comment:
Because I need a minimal one to allow me to test low level functionality on a small graph where I know the (possible) results. Other parts of the code are using this. I just did the same (for example: tests/linear_assignment/hungarian_test.cu)

Reply:
No, you don't need an empty fixture (or a fixture at all) to do what you said.  I do recommend using a fixture with a `SetUp()` in your case, but you don't need an empty one if you have no intention of using it - you'd just use a `TEST` macro instead.

I don't think we should propagate patterns from other files if we can start improving things.

Reply:
The struct name `RandomWalksPrimsTest` is further used in subsequent tests (`TEST_F(RandomWalksPrimsTest, ...)`, etc.) How can it compile without it?

Can you please provide a code snippet example of what you'd like here?

Reply:
Resolving this per latest FIXME comments.